Debian-HA Patches

rikkotec opened the pull request:
@beekhof 

Here are those spelling, and manual file formatting fixes that I (devrikx) mentioned in clusterlabs' IRC.

Feel free to merge or cherry-pick them at your discretion :)

(Failed build on travis-ci seems to be unrelated to any of my changes - and instead because it was unable to download archives from ubuntu)

beekhof commented on tools/crm_attribute.c:
```diff
@@ -125,7 +125,7 @@ main(int argc, char **argv)
 
     crm_log_cli_init("crm_attribute");
-    crm_set_options(NULL, "command -n attribute [options]", long_options,
+    crm_set_options(NULL, "command \\-n attribute [options]", long_options,
```
Whats this one about?

beekhof replied:
Oh I see. Wont this mess up the --help output though?

rikkotec commented:
@beekhof 

Hmm, you are right that the output shows the forward slash:

```shell
Usage: crm_attribute command \-n attribute [options]
Options:
 -?, --help		This text
 -$, --version		Version information
 -V, --verbose		Increase debug output
 -q, --quiet		Print only the value on stdout

 -n, --name=value	Name of the attribute/option to operate on

Commands:
 -G, --query		Query the current value of the attribute/option
 -v, --update=value	Update the value of the attribute/option
 -D, --delete		Delete the attribute/option

Additional Options:
 -N, --node=value	Set an attribute for the named node (instead of a cluster option).  See also: -l
 -t, --type=value	Which part of the configuration to update/delete/query the option in.
			Valid values: crm_config, rsc_defaults, op_defaults, tickets
 -l, --lifetime=value	Lifetime of the node attribute.
			Valid values: reboot, forever
 -z, --utilization	Set an utilization attribute for the node.
 -s, --set-name=value	(Advanced) The attribute set in which to place the value
 -i, --id=value		(Advanced) The ID used to identify the attribute
 -d, --default=value	(Advanced) The default value to display if none is found in the configuration

Examples:

Add a new attribute called 'location' with the value of 'office' for host 'myhost':

	# crm_attribute --node myhost --name location --update office

Query the value of the 'location' node attribute for host myhost:

	# crm_attribute --node myhost --name location --query

Change the value of the 'location' node attribute for host myhost:

	# crm_attribute --node myhost --name location --update backoffice

Delete the 'location' node attribute for the host myhost:

	# crm_attribute --node myhost --name location --delete

Query the value of the cluster-delay cluster option:

	# crm_attribute --type crm_config --name cluster-delay --query

Query the value of the cluster-delay cluster option. Only print the value:

	# crm_attribute --type crm_config --name cluster-delay --query --quiet


Report bugs to pacemaker@oss.clusterlabs.org
```

Though perhaps it may be better to change:

```shell
crm_attribute command -n attribute [options]
```

to 

```shell
crm_attribute command [options] attribute [additional options]
```

What do you think of that? There is no error with the existing proposed fix...but it does show the slash in the output...this other idea actually properly references how one should format the command?

rikkotec commented:
@beekhof 

Tested good:

```shell
root@mmasx:/home/mmod/ha-build/installable-debs# attrd_updater --help
attrd_updater - Tool for updating cluster node attributes
Usage: attrd_updater command [options] attribute [additional options]
Options:
 -?, --help		This text
 -$, --version		Version information
 -V, --verbose		Increase debug output

 -q, --quiet		Print only the value on stdout

 -n, --name=value	The attribute's name

Commands:
 -U, --update=value	Update the attribute's value in attrd.  If this causes the value to change, it will also be updated in the cluster configuration
 -Q, --query		Query the attribute's value from attrd
 -D, --delete		Delete the attribute in attrd.  If a value was previously set, it will also be removed from the cluster configuration
 -R, --refresh		(Advanced) Force the attrd daemon to resend all current values to the CIB


Additional options:
 -l, --lifetime=value	Lifetime of the node attribute.  Allowed values: forever, reboot
 -d, --delay=value	The time to wait (dampening) in seconds further changes occur
 -s, --set=value	(Advanced) The attribute set in which to place the value

Report bugs to pacemaker@oss.clusterlabs.org
```

and

```shell
root@mmasx:/home/mmod/ha-build/installable-debs# crm_attribute --help
crm_attribute - Manage node's attributes and cluster options.

Allows node attributes and cluster options to be queried, modified and deleted.

Usage: crm_attribute command [options] attribute [additional options]
Options:
 -?, --help		This text
 -$, --version		Version information
 -V, --verbose		Increase debug output
 -q, --quiet		Print only the value on stdout

 -n, --name=value	Name of the attribute/option to operate on

Commands:
 -G, --query		Query the current value of the attribute/option
 -v, --update=value	Update the value of the attribute/option
 -D, --delete		Delete the attribute/option

Additional Options:
 -N, --node=value	Set an attribute for the named node (instead of a cluster option).  See also: -l
 -t, --type=value	Which part of the configuration to update/delete/query the option in.
			Valid values: crm_config, rsc_defaults, op_defaults, tickets
 -l, --lifetime=value	Lifetime of the node attribute.
			Valid values: reboot, forever
 -z, --utilization	Set an utilization attribute for the node.
 -s, --set-name=value	(Advanced) The attribute set in which to place the value
 -i, --id=value		(Advanced) The ID used to identify the attribute
 -d, --default=value	(Advanced) The default value to display if none is found in the configuration

Examples:

Add a new attribute called 'location' with the value of 'office' for host 'myhost':

	# crm_attribute --node myhost --name location --update office

Query the value of the 'location' node attribute for host myhost:

	# crm_attribute --node myhost --name location --query

Change the value of the 'location' node attribute for host myhost:

	# crm_attribute --node myhost --name location --update backoffice

Delete the 'location' node attribute for the host myhost:

	# crm_attribute --node myhost --name location --delete

Query the value of the cluster-delay cluster option:

	# crm_attribute --type crm_config --name cluster-delay --query

Query the value of the cluster-delay cluster option. Only print the value:

	# crm_attribute --type crm_config --name cluster-delay --query --quiet


Report bugs to pacemaker@oss.clusterlabs.org
```

This is the latest patch we're using to resolve the related lintian-errors. Hope that helps!

beekhof commented:
I've merged everything except the minus sign patches.  Its not clear to me what benefit they provide (other than satisfying the arbitrary requirements of some tool).

rikkotec commented:
On Sun, 2015-05-10 at 17:34 -0700, Andrew Beekhof wrote:
> I've merged everything except the minus sign patches. Its not clear to
> me what benefit they provide (other than satisfying the arbitrary
> requirements of some tool).

I believe it is a lintian informational warning that can be safely
ignored [1], those two patches were mainly for covering all ends, if it
were to have been desired; I do see now that there is no harm with how
those hyphens are present in the respective manuals

Thanks for the merge!


References: 

[1] https://lintian.debian.org/tags/hyphen-used-as-minus-sign.html


Best,


-- 
Rik

beekhof commented:
Closing this one now that we've decided which ones we're going to merge (and have done so)
